lib/chef/provider/apt_repository.rb

```diff
@@ -206,7 +206,7 @@ def install_key_from_uri(key)
           mode "0644"
           sensitive new_resource.sensitive
           action :create
-          verify "gpg %{path}"
+          verify "gpg --homedir /tmp/ %{path}"
```

Comment:
We probably want to use tempfile here so we can get a unique path to the file. https://ruby-doc.org/stdlib-1.9.3/libdoc/tempfile/rdoc/Tempfile.html

Reply:
yes we can use tempfile here and I even tried using it but as I thought it will not maintain idempotancy as tempfiles get deleted automatically when ruby interpreter shuts down.

Reply:
Will that impact the idempotency of the resource? It should just recreate the .gpg directory on each Chef run, which seems fine and ideal since we'd avoid reading in a predictable directory that is globally writable.